### PR TITLE
Minor Fine-Tuning Scheduler Tutorial Updates for PTL 1.7

### PR DESCRIPTION
**Pull request by speediedan**
## Before submitting
- [X] Did you make sure to **update the docs**?
- [X] Did you write any new **necessary tests**?

## What does this PR do?

A few minor updates to the Fine-Tuning Scheduler tutorial to better support the release of pytorch-lightning 1.7.x

- changed "finetuning" uses to "fine-tuning" since closed compound word version remains less common in the research literature for the time being
- removed use of CLI registry since it is deprecated in PTL 1.7 and simplifies the tutorial
- removed hydra-core version requirement since the relevant issue has been resolved
- added a small patch to address [#80809](https://github.com/pytorch/pytorch/issues/80809) and allow the tutorial to succeed with torch 1.12.0. Despite the recent release of 1.12.1, the tutorial should work with 1.12.0 since a non-trivial number of users will likely continue to have it installed. 

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by speediedan** [comment]
FYI, the validate Docs/ build-docs job is only failing due to a [single stale pytorch-lightning link on the basic_gan nb](https://github.com/Lightning-AI/tutorials/runs/7709035044?check_suite_focus=true#step:11:673)

**Review by tchaton — APPROVED** [review]
Looks neat !

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py [hunk 1]
```diff
@@ -160,68 +160,59 @@
...
-    if not require_fqn:
-        for n, c in MOCK_REGISTRY.items():
-            globals()[f"{n}"] = c
-        registered_list = ", ".join([n for n in MOCK_REGISTRY.names])
-    else:
-        registered_list = ", ".join([c.__module__ + "." + c.__name__ for c in MOCK_REGISTRY.classes])
-    print(f"Imported and registered the following callbacks: {registered_list}")
-
+if Version(torch.__version__) == Version("1.12.0") or torch.__version__.startswith("1.12.0"):
+    # we need to use a patched version of AdamW to fix https://github.com/pytorch/pytorch/issues/80809
```
Comment (rohitgr7) [inline lightning_examples/finetuning-scheduler/finetuning-scheduler.py:191]: > https://github.com/pytorch/pytorch/issues/80809

this must have broken a lot of code .. haha

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py [hunk 2]
```diff
@@ -376,25 +367,22 @@ def __init__(
...
+        return fts_callback[0] if fts_callback else None
 
     def forward(self, **inputs):
         return self.model(**inputs)
 
     def training_step(self, batch, batch_idx):
         outputs = self(**batch)
         loss = outputs[0]
-        self.log("train_loss", loss)
+        self.log("train_loss", loss, sync_dist=True)
```
Comment (rohitgr7) [inline lightning_examples/finetuning-scheduler/finetuning-scheduler.py:380]: this won't have any effect unless you do `on_epoch=True` in `training_step` which is `False` by default here.

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py [hunk 3]
```diff
@@ -451,20 +439,20 @@ def configure_optimizers(self):
...
+# upon a shared set of defaults, only differing in their respective fine-tuning schedules.
 #
 # | Experiment Tag    | Training Scenario Description                                          |
 # |:-----------------:| ---------------------------------------------------------------------- |
-# | ``fts_explicit``  | Training with a finetuning schedule explicitly provided by the user    |
-# | ``nofts_baseline``| A baseline finetuning training session (without scheduled finetuning)  |
-# | ``fts_implicit``  | Training with an implicitly generated finetuning schedule (the default)|
+# | ``fts_explicit``  | Training with a fine-tuning schedule explicitly provided by the user    |
+# | ``nofts_baseline``| A baseline fine-tuning training session (without scheduled fine-tuning)  |
+# | ``fts_implicit``  | Training with an implicitly generated fine-tuning schedule (the default)|
```
Comment (rohitgr7) [inline lightning_examples/finetuning-scheduler/finetuning-scheduler.py:449]:
```suggestion
# | ``fts_explicit``  | Training with a fine-tuning schedule explicitly provided by the user     |
# | ``nofts_baseline``| A baseline fine-tuning training session (without scheduled fine-tuning)  |
# | ``fts_implicit``  | Training with an implicitly generated fine-tuning schedule (the default) |
```

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py [hunk 4]
```diff
@@ -160,68 +160,59 @@
...
-    # make registered class available by unqualified class name by default
-    if not require_fqn:
-        for n, c in MOCK_REGISTRY.items():
-            globals()[f"{n}"] = c
-        registered_list = ", ".join([n for n in MOCK_REGISTRY.names])
-    else:
-        registered_list = ", ".join([c.__module__ + "." + c.__name__ for c in MOCK_REGISTRY.classes])
-    print(f"Imported and registered the following callbacks: {registered_list}")
-
+if Version(torch.__version__) == Version("1.12.0") or torch.__version__.startswith("1.12.0"):
```
Comment (rohitgr7) [inline lightning_examples/finetuning-scheduler/finetuning-scheduler.py:190]: what's up with `startswith(1.12.0)`? like this version will be that?

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py [hunk 5]
```diff
@@ -160,68 +160,59 @@
...
+    # we need to use a patched version of AdamW to fix https://github.com/pytorch/pytorch/issues/80809
+    # and allow examples to succeed with torch 1.12.0 (this torch bug is fixed in 1.12.1)
+    from fts_examples.patched_adamw import AdamW
+else:
+    from torch.optim.adamw import AdamW
```
Comment (rohitgr7) [inline lightning_examples/finetuning-scheduler/finetuning-scheduler.py:195]: or let's use `SGD`? since it's just an example.

**Review by rohitgr7 — APPROVED** [review]

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py (hunk 1 above)
Comment (speediedan) [reply lightning_examples/finetuning-scheduler/finetuning-scheduler.py:191]: 😂 omg, a huge pain for them I'm sure, but I definitely have sympathy and they got the minor patch out surprisingly fast relative to previous release minor patches (probably an indicator of how problematic it was!)

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py (hunk 2 above)
Comment (speediedan) [reply lightning_examples/finetuning-scheduler/finetuning-scheduler.py:380]: good point, thanks! I over-zealously applied these ``sync_dist``s as a workaround when I was debugging a separate multi-gpu ddp_spawn issue but it is not relevant in this case and it definitely shouldn't be included in a tutorial!

Comment (speediedan) [reply lightning_examples/finetuning-scheduler/finetuning-scheduler.py:380]: I'll push another commit shortly with the ``sync_dist``s removed

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py (hunk 4 above)
Comment (speediedan) [reply lightning_examples/finetuning-scheduler/finetuning-scheduler.py:190]: I actually didn't recognize the need for this condition set extension until I installed ``torch`` using this method below for some other testing/dev purposes:
```bash
pip install torch torchvision torchtext --extra-index-url https://download.pytorch.org/whl/cu116;
python -c "import torch;from packaging.version import Version; print(f\"torch version installed: {torch.__version__}\"); print(f\"Version test: {Version(torch.__version__) == Version('1.12.1')}\"); print(f\"startswith test: {torch.__version__.startswith('1.12.1')}\");"
torch version installed: 1.12.1+cu116
Version test: False
startswith test: True
```
Unfortunate ``Version`` doesn't recognize that version style but I get it, interesting anyway, every day has its own tutorial :smile: ! I included both test conditions just in case.

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py (hunk 5 above)
Comment (speediedan) [reply lightning_examples/finetuning-scheduler/finetuning-scheduler.py:195]: If it's okay, I'd prefer to stick with ``AdamW`` despite the bug since I use that optimizer in other examples and it might be nicer to have things as similar as possible.

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py [hunk 6]
```diff
@@ -160,68 +160,59 @@
...
+for hflogger in [transformers_logging, datasets_logging]:
+    hflogger.set_verbosity_error()
 # ignore warnings related tokenizers_parallelism/DataLoader parallelism trade-off and
 # expected logging behavior
-for warnf in [".*does not have many workers*", ".*The number of training samples.*"]:
+for warnf in [
+    ".*does not have many workers.*",
+    ".*The number of training samples.*",
+    ".*converting to a fast.*",
+    ".*number of training batches.*",
```
Comment (Borda) [inline lightning_examples/finetuning-scheduler/finetuning-scheduler.py:214]:
```suggestion
    r".*does not have many workers.*",
    r".*The number of training samples.*",
    r".*converting to a fast.*",
    r".*number of training batches.*",
```
Shall be like this?

**Review by Borda — APPROVED** [review]

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py (hunk 5 above)
Comment (rohitgr7) [reply lightning_examples/finetuning-scheduler/finetuning-scheduler.py:195]: cool!

File: lightning_examples/basic-gan/gan.py [hunk 7]
```diff
@@ -20,7 +20,7 @@
 # ### MNIST DataModule
 #
 # Below, we define a DataModule for the MNIST Dataset. To learn more about DataModules, check out our tutorial
-# on them or see the [latest docs](https://pytorch-lightning.readthedocs.io/en/stable/extensions/datamodules.html).
+# on them or see the [latest release docs](https://pytorch-lightning.readthedocs.io/en/stable/data/datamodules.html).
```
Comment (rohitgr7) [inline lightning_examples/basic-gan/gan.py:23]:
```suggestion
# on them or see the [latest release docs](https://pytorch-lightning.readthedocs.io/en/stable/data/datamodule.html).
```

**Review by Borda — APPROVED** [review]
